spec/ci_helper.rb

```diff
@@ -0,0 +1,57 @@
+require "linux_admin"
+
+require_relative 'support/mixins/pg_environment_updater'
+
+class CiPostgresRunner
+  PGVER       = "9.5".freeze
+  PGDATADIR   = "/var/lib/postgresql/#{PGVER}/main".freeze
+  PGCONFIGDIR = "/etc/postgresql/#{PGVER}/main".freeze
+
+  def self.start
+    # Make sure we have our postgresql.conf in the right spot on Travis
+    system("cp #{PGDATADIR}/postgresql.conf #{PGCONFIGDIR}/postgresql.conf")
+
+    # Move our configured `pg_hba.conf` to the config dir as well
+    system("cp #{PGDATADIR}/pg_hba.conf #{PGCONFIGDIR}/pg_hba.conf")
```

Comment:
Why do we need to do these? I would think the built-in configuration would work for this.

Reply:
This is something I really needed document better in the PR description... and probably in the code.

Travis uses a Debian image, and they use a filesystem (feature?) called RAMFS, which I assume is to improve performance on boot or some other performance reason.  From what I found, it still ends up pointing back to the data directory in some fashion, but these files need to exist in the `/var/ramfs/...` dir on first boot for the `postgres` process to read that, instead of the one that existed previously (which wouldn't be a great example of a restore if we didn't).

Again, possibly using the `pg_ctl` method would have been wiser for CI...

Reply:
Because our appliance configuration actually sets `ssl = on`, and I figured that would be disingenuous if we didn't try reconnecting with SSL through the port connection.

Reply:
Oh I see, this is not "our" configuration it's the config provided by the built-in PG on travis.

I misunderstood. I thought you were somehow grabbing the conf files that we install on appliances and trying to use those.

Reply:
> Oh I see, this is not "our" configuration it's the config provided by the built-in PG on travis.

No, you were right with your first assumption (and clearly I shouldn't be responding to code review in the morning...).  Let me see if I can do a better job explaining...

In "our" setup, we have the database directory and the config directory in the same spot, so when we do a database restore by just unziping the tar, everything is ready to go.  On Travis, because there is a split between what is configured as the data directory and what is the config directory, this `cp` of files is necessary, otherwise we are booting `postgres` with old configs, which really isn't how we assume our restores work.

Reply:
Oh so this is happening after the restore. That makes more sense. This seems like something else that could be solved by starting up a server using pg_ctl in both cases as well. So this can go into the list of things to tackle in that follow-up refactoring.